src/Middleware/Microsoft.AspNetCore.OutputCaching.StackExchangeRedis/src/RedisCacheOptions.cs

```diff
@@ -0,0 +1,76 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Options;
+using StackExchange.Redis;
+using StackExchange.Redis.Configuration;
+using StackExchange.Redis.Profiling;
+
+namespace Microsoft.AspNetCore.OutputCaching.StackExchangeRedis;
+
+/// <summary>
+/// Configuration options for Redis based output cache.
+/// </summary>
+public class RedisOutputCacheOptions : IOptions<RedisOutputCacheOptions>
```

Comment:
Is there a way we can depend on the existing StackExchangeRedis package and share/derive from its options type and otherwise reduce duplication?

Reply:
In theory, yes. However, it seems an odd choice to take a package dependency *just* for a config definition.

Reply:
If I wanted a Distributed Cache and an Output Cache in my app that both talked to the same redis server, do I need to now configure them both the same way?

I wonder if we could put `ConfigurationOptions` as the thing that is an IOptions that I can configure. That way apps can configure redis distributed caches and redis output caches the same way and they compose together.

Reply:
It seems like a maintenance and config issue. E.g. having to add, implement, and configure options in both places.

I don't quite follow your suggestion @eerhardt. It would be nice to configure the two together somehow.

Reply:
> I don't quite follow your suggestion

My suggestion is to use the `StackExchange.Redis.ConfigurationOptions` type as the "Options" object. That is what is configured in DI instead of these separate `RedisOutputCacheOptions` and `RedisCacheOptions` objects which have properties of type `StackExchange.Redis.ConfigurationOptions`.

One issue with IOptions is that they don't compose together well. You can't have have 2 IOptions objects that refer to the same instance without manually writing code to wire them together.

Reply:
@Tratcher  I 100% agree on the configure together thing; *an* option there is to use the option that provides the multiplexer directly, although that then doesn't have quite the same capability re recreating the connection; I suggest we tackle this in net9, though